### ISSUE TITLE
GenericWhitespace check does not recognize java 8 method references 

### ISSUE THREAD

**Issue by Zingler** [checkstyle/checkstyle#677]
As per http://checkstyle.sourceforge.net/config_whitespace.html, the right angle bracket should not be followed by whitespace when adjacent to a method name. In checkstyle 6.3, using a method reference with generics results in the following error.  

'>' is followed by an illegal character.

Issue can be reproduced using the GenericWhitespace check on the following line.
```
final Supplier<Optional<Integer>> function = Optional::<Integer>empty;
```
while it passes on
```
final Supplier<Optional<Integer>> function = Optional::<Integer> empty;
```
Which I do not think is the intended behavior.

**Comment by romani** [comment]
merged, will be in 6.5

**Comment by romani** [comment]
released as 6.4.1
